ServiceBrokerListener/ServiceBrokerListener.Domain/SqlDependencyEx.cs

```diff
@@ -257,7 +257,8 @@ FROM DELETED
                 DEALLOCATE Conv;
 
                 -- Droping service and queue.
-                DROP SERVICE [{1}];
+                IF @serviceId >= 0
```

Comment:
This is a great catch, thank you!
But better thing here would be:
`IF (@serviceId IS NULL)`